easy-table-of-contents.php

```diff
@@ -779,7 +780,8 @@ public static function extract_headings( &$find, &$replace, $content = '' ) {
 
 								$items .= '<li><a href="' . esc_url( '#' . $anchor ) . '">';
 								//$title  = apply_filters( 'ez_toc_title', strip_tags( wp_kses_post( $toc[ $i ][0] ) ) );
-								$title  = strip_tags( apply_filters( 'ez_toc_title', $toc[ $i ][0] ) );
+								$title = apply_filters( 'ez_toc_title', $matches[ $i ][0] );
+								$title = preg_replace( '/^<h[^>]*>|<\/h\d>$/muU', '', $title );
```

Comment:
Perhaps this regex?

```php
preg_replace( '/^<h([1-6])[^>]*>\s?(.*)?\s?<\/h\1>$/uUmsi', '$2', $title );
```

Reply:
It would seem sensible to take into account the expression that generated the title in the first place, which is `'/(<h([1-6]{1})[^>]*>).*<\/h\2>/msuU'`. I actually started out with that expression and got something very similar to what you are suggesting, but I changed my mind when I realized that this regular expression actually specifies the format I'd have to parse and I can therefore simplify my expression. In fact, I'd like to simplify it even more. Since we're not expecting headers within headers, we could go with a super simple `'/<\/?h[^>]*>/muU'`. That would be my favorite, as reflected by my latest commit.

Also, I see you have added an 'i' modifier. Uppercase tags are not currently supported by the regular expression that generates the title (and are not supported by TOC+ either). If you'd like to add support for uppercase tags, you'd have to update the other regular expression as well.

Reply:
After reviewing your updated code I had actually started with a more simple regex, similar to the you ended with and went more complex with it in mind that if the code will be refactored it would make that section more portable... but that'll probably never will happen ;)

Your changes will not break existing functionality so, merged! :)

Reply:
:D There are pros and cons either way. I'm happy my `<wbr>` tags are no longer stripped.